backend/tests/test_indexing.py

```
@@ -198,8 +198,8 @@ def side_effect_write(index, filepath):
     @patch('faiss.read_index')
     @patch('os.path.exists')
     @patch('builtins.open')
-    @patch('pickle.load')
-    def test_load_index_preserves_data(self, mock_pickle_load, mock_open, mock_exists, mock_read_index):
+    @patch('json.load')
+    def test_load_index_preserves_data(self, mock_json_load, mock_open, mock_exists, mock_read_index):
```

Ruff (0.15.1)

[warning] Unused method argument: `mock_open`

(ARG002)